src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -1076,12 +1077,19 @@ public static Set<PIP> getTrimmablePIPsFromPins(Net net, Collection<SitePinInst>
 	            rPips.add(pip);
 	        }
 
-	        Integer count = fanout.get(startNode);
-	        if(count == null){
-	            fanout.put(startNode, 1);
-	        }else{
-	            fanout.put(startNode, count+1);
+	        int fanoutCount = 1;
+	        SitePin sp = startNode.getSitePin();
```

Comment:
I was thinking about this some more, and `Node.getSitePin()` can be expensive.  I'm thinking about an alternate solution.  Doesn't have to block this, but I think we might see some slow down with this approach as it is called a fair amount in the regressions.

Reply:
Since we know the net upfront and we are relying on the `SitePinInst` objects to be present, we can quickly build a map upfront of all the nodes connected to the `SitePinInst` of the net.  Then, we simply perform a hash lookup into the map to see if the node of interest has a pin on it.

Reply:
Take a look at https://github.com/Xilinx/RapidWright/pull/474